src/java.base/share/classes/java/text/ListFormat.java

```diff
@@ -0,0 +1,591 @@
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+package java.text;
+
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.Serial;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.regex.Pattern;
+import java.util.stream.IntStream;
+import sun.util.locale.provider.LocaleProviderAdapter;
+
+/**
+ * {@code ListFormat} produces or parses a list of concatenated
+ * strings in a locale-sensitive way. Use {@code ListFormat} to construct a list
+ * of strings displayed for end users. For example, displaying 3 weekdays
+ * as "Monday, Wednesday, and Friday". This class provides the functionality
+ * defined in Unicode Consortium's LDML specification for
+ * <a href="https://www.unicode.org/reports/tr35/tr35-general.html#ListPatterns">
+ * List Patterns</a>.
+ * <p>
+ * Three types of concatenation are provided: {@link Type#STANDARD STANDARD},
+ * {@link Type#OR OR}, and {@link Type#UNIT UNIT}, also three styles for each
+ * type are provided: {@link Style#FULL FULL}, {@link Style#SHORT SHORT}, and
+ * {@link Style#NARROW NARROW}. The following snippet is an example of formatting
+ * the list of Strings {@code "Foo", "Bar", "Baz"} in US English with
+ * {@code STANDARD} type and {@code FULL} style:
+ * {@snippet lang=java :
+ * ListFormat.getInstance(Locale.US, ListFormat.Type.STANDARD, ListFormat.Style.FULL)
+ *     .format(List.of("Foo", "Bar", "Baz"))
+ * }
+ * This will produce the concatenated list string, "Foo, Bar, and Baz" as seen in
+ * the following table:
+ * <table class="striped">
+ * <caption style="display:none">Formatting examples</caption>
+ * <thead>
+ * <tr><th scope="col"></th>
+ *     <th scope="col">FULL</th>
+ *     <th scope="col">SHORT</th>
+ *     <th scope="col">NARROW</th></tr>
+ * </thead>
+ * <tbody>
+ * <tr><th scope="row" style="text-align:left">STANDARD</th>
+ *     <td>Foo, Bar, and Baz</td>
+ *     <td>Foo, Bar, &amp; Baz</td>
```

Comment:
Is "&amp;" a typo?  It's still "and" in the Unicode spec's "standard-short" format, e.g. "Jan., Feb., and Mar."

Reply:
Again `ampersand` is the correct pattern for `SHORT` in English.